### PR TITLE
bt-558: Employer profile page

### PR DESCRIPTION
**Pull request by ptarho**
User can open this page by clicking on profile icon in chat header.
If he will try to mannualy enter this page with non-existing company id he will be redirected to 404 page
Screenshot_1
Screenshot_2

### CODE REVIEW

File: frontend/src/bundles/employers/types/employers-data-dto.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+type EmployerDataDto = {
+    companyWebsite: string;
+    fullName: string;
+    companyLogo: { url: string };
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/employers/types/employers-data-dto.ts]: I think it should be like general File type like { id: string, url: string ... } or something like that

File: frontend/src/bundles/employers/pages/company/company-page.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                const data = await employersApi.getEmployerDetails(
+                    id as string,
+                );
+                setCompanyData(data);
+            } catch {
+                navigate('/404');
+            }
+        }
+        void fetchCompanyData();
+    }, [id, navigate]);
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/employers/pages/company/company-page.tsx]: I think we can use redux for this ?

File: backend/src/bundles/user-details/user-details.entity.ts [hunk 3]
```diff
@@ -66,6 +67,10 @@ class UserDetailsEntity implements Entity {
 
     private 'completedStep': ValueOf<typeof OnboardingSteps> | null;
 
+    private 'photo': ValueOf<typeof FileModel> | null;
+
+    private 'companyLogo': ValueOf<typeof FileModel> | null;
+
```
Comment (MaksymGavatuk) [inline backend/src/bundles/user-details/user-details.entity.ts]: We don't need this field in entity

File: backend/src/bundles/user-details/user-details.model.ts [hunk 4]
```diff
@@ -83,9 +83,14 @@ class UserDetailsModel extends AbstractModel {
 
     public 'completedStep': ValueOf<typeof OnboardingSteps>;
 
+    public 'photo': ValueOf<typeof FileModel> | null;
+
+    public 'companyLogo': ValueOf<typeof FileModel> | null;
```
Comment (MaksymGavatuk) [inline backend/src/bundles/user-details/user-details.model.ts]: Why not just FileModel | null, ... ?

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/employers/types/employers-data-dto.ts (hunk 1 above)
Comment (ptarho) [reply frontend/src/bundles/employers/types/employers-data-dto.ts]: Fixed

File: frontend/src/bundles/employers/pages/company/company-page.tsx (hunk 2 above)
Comment (ptarho) [reply frontend/src/bundles/employers/pages/company/company-page.tsx]: Fixed

File: backend/src/bundles/user-details/user-details.entity.ts (hunk 3 above)
Comment (ptarho) [reply backend/src/bundles/user-details/user-details.entity.ts]: Fixed

File: backend/src/bundles/user-details/user-details.model.ts (hunk 4 above)
Comment (ptarho) [reply backend/src/bundles/user-details/user-details.model.ts]: Fixed

**Comment by nikita-remeslov** [comment]
I think we should make the design similar to the candidate page
image

**Pull request closed without merging**